.github/workflows/reusable-pr.yml

```diff
@@ -96,18 +102,28 @@ jobs:
             esac
             sed -i -e "s|KUSTOMIZE_DIR_REPLACE|$kustomiseDir|g" secure-api-gateway-ci/component-config/${{ inputs.componentName }}.env
           fi
-          # Override snapshot version if artifacts are being uploaded
+          # We are uploading some build artifacts to maven with a [VERSION]-[PR_NUMBER]-SNAPSHOT format, need to override SNAPSHOT_REPLACE in the env file
           if grep -q SNAPSHOT_REPLACE secure-api-gateway-ci/component-config/${{ inputs.componentName }}.env; then
             # Create the snapshot version '[VERSION]-[PR_NUMBER]-SNAPSHOT'
+            # Get the current version from the pom.xml file
             pomVersion=$(cd ${{ inputs.componentName }} && mvn help:evaluate -Dexpression=revision -q -DforceStdout)
+            # Remove SNAPSHOT from the retrieved version if it exists
             if [[ "$pomVersion" =~ "-SNAPSHOT" ]]; then
               pomVersion=${pomVersion/-SNAPSHOT/""}
             fi
+            # Build the new version from the retrieved version and docker tag and add snapshot to the end
             pomVersion="$pomVersion-${{ inputs.dockerTag}}-SNAPSHOT"
+            # Override SNAPSHOT_REPLACE in the env file with the built version
             sed -i -e 's/SNAPSHOT_REPLACE/'$pomVersion'/g' secure-api-gateway-ci/component-config/${{ inputs.componentName }}.env
             echo STEPS_PR_OUTPUT_SNAPSHOT_VERSION="true" >> $GITHUB_ENV
             echo STEPS_PR_SNAPSHOT_VERSION=$pomVersion >> $GITHUB_ENV
           fi
+          # If there are maven args specified from the inputs, then we need to override MAVEN_ARG_OVERRIDE in the env file
+          if [[ -n "$mavenArg" ]]; then
+            if grep -q MAVEN_ARG_OVERRIDE secure-api-gateway-ci/component-config/${{ inputs.componentName }}.env; then
+              sed -i -e 's/MAVEN_ARG_OVERRIDE/'mavenArgs=\"-Dopenig.version=${{ inputs.mavenArg }}\"'/g' secure-api-gateway-ci/component-config/${{ inputs.componentName }}.env
+            fi
+          fi
```

Comment:
I think this is breaking my PR workflow for core.

The logic should check if the `MAVEN_ARG_OVERRIDE` exists first, then if $mavenArg exists use that to do the replacement otherwise replace with empty string (to delete MAVEN_ARG_OVERRIDE from the command).

See build: https://github.com/SecureApiGateway/secure-api-gateway-core/actions/runs/11610565997/job/32330096462